### DISCUSSION TITLE
Add "Complex" Callout to Markdown output

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#6604] [Feature Requests]
### Description

[Callout blocks](https://quarto.org/docs/authoring/callouts.html) are first render into three formats (CommonMark, HTML, and ePub), see https://github.com/quarto-dev/quarto-cli/blob/eafea4c58ce329ee153f39be7ff0c43e0f900313/src/resources/filters/customnodes/callout.lua#L567-L576

Would be great to have simplified HTML, something like

```
<div class="callout callout-important">
<div class="callout-header">
Important
</div>
<div class="callout-body">
<p>This is important!</p>
</div>
</div>
```

that authors can style with beginer level CSS.

I believe this can be done by changing the previous code snippet to

```
-- default renderer first
_quarto.ast.add_renderer("Callout", function(_)
  return true
end, simpleCallout)
_quarto.ast.add_renderer("Callout", function(_)
  return _quarto.format.isHtmlCompatible()
end, simpleCalloutDiv)
_quarto.ast.add_renderer("Callout", function(_)
  return _quarto.format.isHtmlOutput() and hasBootstrap()
end, calloutDiv)
_quarto.ast.add_renderer("Callout", function(_) 
  return _quarto.format.isEpubOutput() or _quarto.format.isRevealJsOutput()
end, epubCallout)
```

and writing the `simpleCalloutDiv` function. Is a PR welcome?

**Comment by cscheid** [comment]
This is an interesting idea and PRs are always welcome! But this needs to be handled delicately. A few issues:

1) Immediately, I think your suggestion will break `revealjs` rendering, because it is html-compatible. 

2) If we create this, then we're promising to support that DOM effectively forever, which means we need to make sure it does the right thing. (For example, I'm not really willing to compromise on a11y here, and I'm not sure that DOM does the right thing).

3) Formats and rendering order are fairly fragile; I'm not sure if our test suite exercises well the difference between these different HTML-compatible formats. It should, but it doesn't, so we need to be careful there.

This is not exhaustive, but gives you an idea for what we need to solve before merging an eventual PR.

**Reply by rgaiacs** [reply]
Thanks for the feedback.
